File: Programmers/과일장수.py
def solution(k, m, score):
    answer = 0
    score.sort(reverse=True)
    box = []
    temp = []
    cnt = 0
  
    for i in score:
        temp.append(i)
        cnt += 1
        if cnt%m == 0:
            box.append(temp)
            cnt = 0
            temp = []
    for i in box:
        answer += min(i)*m
      
    return answer

File: Programmers/과일장수.c
#include <stdio.h>
#include <stdlib.h>

int compare(const void *a, const void *b) {
    return (*(int*)b - *(int*)a);
}

int solution(int k, int m, int* score, int scoreSize) {
    int answer = 0;
    
    qsort(score, scoreSize, sizeof(int), compare);

    int cnt = 0;
    int temp[m];
    
    for (int i = 0; i < scoreSize; i++) {
        temp[cnt++] = score[i];
        if (cnt == m) {
            int minVal = temp[0];
            for (int j = 1; j < m; j++) {
                if (temp[j] < minVal) {
                    minVal = temp[j];
                }
            }
            answer += minVal * m;
            cnt = 0;
        }
    }
    
    return answer;
}
